dbms/src/Storages/StorageHDFS.cpp

```diff
@@ -59,7 +61,15 @@ class HDFSBlockInputStream : public IBlockInputStream
         const Context & context,
         UInt64 max_block_size)
     {
-        std::unique_ptr<ReadBuffer> read_buf = std::make_unique<ReadBufferFromHDFS>(uri);
+        std::unique_ptr<ReadBuffer> read_buf;
+        if (boost::algorithm::ends_with(uri, ".gz")) {
```

Comment:
Better to use https://github.com/clickhouse/ClickHouse/blob/master/dbms/src/Common/StringUtils/StringUtils.h#L34.

Reply:
I'm not sure, that the decision of compression type should be made on the file extension. Maybe better to add an optional parameter to`StorageHDFS` with compression name?

Reply:
> I'm not sure, that the decision of compression type should be made on the file extension. Maybe better to add an optional parameter to`StorageHDFS` with compression name?

Why not both? :)
I will add an optional parameter, but I also want to keep the option to choose a compression method by file extension.